### PR TITLE
Fix non-32-bit clz

### PR DESCRIPTION
**Pull request by alan-baker**
* Implement non-32-bit clz in terms of 32-bit clz
  * GLSL extended instruction FindUMsb only supports 32-bit integers
* Add amber and unit tests

### CODE REVIEW

**Comment by alan-baker** [comment]
The char and short versions are a little bit ugly, but the way clz is handled in the producer is a bit awkward. Longer term this should be refactored to generate the code a bit cleaner. Probably something like `InsertSPIRVOp` for glsl extended instructions. Also may want to first lower to `llvm.ctlz` and then handle that in llvm intrinsic replacement.

File: amber/run_tests.py [hunk 1]
```diff
@@ -32,7 +32,9 @@
     'images/write_image2d_r32i.amber',
     'images/write_image2d_rg32i.amber',
     'images/write_image2d_rgba32i.amber',
-    'integer/add_sat_short.amber']
+    'integer/add_sat_short.amber',
+    'integer/clz_long.amber',
+    'integer/clz_short.amber']
```
Comment (jrprice) [inline amber/run_tests.py:37]: Missing the `clz_int.amber` script that you added?

File: lib/ReplaceOpenCLBuiltinPass.cpp [hunk 2]
```diff
@@ -2745,3 +2749,58 @@ bool ReplaceOpenCLBuiltinPass::replaceAtomicCompareExchange(Function &F) {
...
+      ty = VectorType::get(ty, vec_ty->getElementCount());
+    }
+    auto clz_32bit_ty = FunctionType::get(ty, {ty}, false);
+    std::string clz_32bit_name = Builtins::GetMangledFunctionName("clz", ty);
+    auto clz_32bit =
+        F.getParent()->getOrInsertFunction(clz_32bit_name, clz_32bit_ty);
+    if (bitwidth < 32) {
+      // Extend the input to 32-bits and perform a clz.  The clz for 32-bit is
+      // tranlated as 31 - FindUMsb(in). Adjust to that result to the right
+      // size.
```
Comment (jrprice) [inline lib/ReplaceOpenCLBuiltinPass.cpp:2776]: Couple of typos here: "tranlated" and "Adjust ~to~ that".

**Review by jrprice — APPROVED** [review]

File: amber/run_tests.py (hunk 1 above)
Comment (alan-baker) [reply amber/run_tests.py:37]: No, those are swiftshader suppressions. Right now swiftshader doesn't support non-32-bit types so only those variants are filtered out.

Comment (jrprice) [reply amber/run_tests.py:37]: Ah sorry, I should've expanded the diff to get that context.

**Pull request merged**